counsel.el

```diff
@@ -3254,6 +3254,36 @@ Note: Duplicate elements of `kill-ring' are always deleted."
  '(("d" counsel-yank-pop-action-remove "delete")
    ("r" counsel-yank-pop-action-rotate "rotate")))
 
+;;** `counsel-evil-registers'
+
+(defcustom counsel-evil-registers-height 5
+  "The `ivy-height' of `counsel-evil-registers'."
+  :group 'ivy
+  :type 'integer)
+
+(defun counsel-evil-registers ()
+  "Ivy replacement for `evil-show-registers'."
+  (interactive)
+  (let ((ivy-format-function #'counsel--yank-pop-format-function)
+        (ivy-height counsel-evil-registers-height))
+    (ivy-read "evil-registers: "
+              (cl-loop for (key . val) in (evil-register-list)
```

Comment:
Perhaps it would be nicer to give an error message at the start of the command if `evil` is not available.

Reply:
Yeah, I agree. That was mainly question number 3.

Maybe I can add a function like (counsel-require-feature) that's similar to the other one... which we can use to gate this function.

What do you think?

Reply:
If you see enough places that follow a "if feature X is unavailable, display an error message explaining this" pattern then sure, I support adding such a convenience function. IIUC, it would take two arguments, the required feature and the message to fail with. But this would belong in a separate PR. If there aren't enough call sites throughout this project, I don't think it would add much value.